### PR TITLE
added action immediatelyResetRouteStack

### PR DESCRIPTION
**Pull request by lucasfeliciano**
#5

### CODE REVIEW

File: reducers/index.js [hunk 1]
```diff
@@ -62,6 +70,17 @@ module.exports = function navigate(state, {type, payload}) {
...
       });
 
+    case IMMEDIATELY_RESET_ROUTE_STACK:
+      invariant(
+        payload && payload.length >= 1,
+        'Make sure to pass an array of routes with at least one route.'
+      );
+
+      return Object.assign({}, state, {
+        stack: new Immutable.Stack(payload),
```
Comment (Kureev) [inline reducers/index.js]:
``` js
state.stack.clear().unshift(payload);
```

^ in this case you wouldn't need to include `immutable` into reducers.

Comment (lucasfeliciano) [inline reducers/index.js]: Good point :+1:

Comment (lucasfeliciano) [inline reducers/index.js]: Actually, doing like you suggested will result in a Stack with the first element as the array passed.

screen shot 2016-02-20 at 19 52 37

Comment (lucasfeliciano) [inline reducers/index.js]:
``` js
state.stack.clear().pushAll(payload);
```

 it is

Comment (lucasfeliciano) [inline reducers/index.js]: done :+1:

**Pull request merged**